src/libmongoc/tests/test-conveniences.h

```diff
@@ -235,8 +243,34 @@ void
 semver_parse (const char *str, semver_t *out);
 
 void
-server_semver (semver_t *out);
+server_semver (mongoc_client_t *client, semver_t *out);
+
+int
+semver_cmp (semver_t *a, semver_t *b);
+
+int
+semver_cmp_str (semver_t *a, const char* str);
 
-int semver_cmp (semver_t *a, semver_t *b);
+const char *
+semver_to_string (semver_t *str);
+
+/* Iterate over a BSON document or array.
+ *
+ * Example of iterating and printing an array of BSON documents:
+ *
+ * bson_t *arr = my_func();
+ * BSON_FOREACH_BEGIN (arr, iter) {
+ *    bson_t el;
+ *    bson_iter_bson (&iter, &el);
+ *    printf ("%d: %s", bson_iter_key (&iter), tmp_json (&el));
+ * }
+ * BSON_FOREACH_END;
+ */
+#define BSON_FOREACH_BEGIN(bson, iter_varname)  \
+   do {                                         \
+      bson_iter_t iter_varname;                 \
+      bson_iter_init (&(iter_varname), (bson)); \
+      while (bson_iter_next (&(iter_varname)))
+#define BSON_FOREACH_END } while (0)
```

Comment:
This is cool! Have you considered removing the `BSON_FOREACH_END` requirement? If you require `iter_varname` to be declared outside of this block too, then you could use a for loop instead of a while loop. In other words:
```
   #define BSON_FOREACH for (bson_iter_init(&iter, bson); bson_iter_next(&iter);) {
```

Reply:
I wavered on it, but I think removing the `BSON_FOREACH_END` and having the iterator declared before seems like a decent trade-off. Done.